### ISSUE TITLE
[java] LawOfDemeter(method chain calls): false negative when using `this.`

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4414]
**Affects PMD Version:** 6.55.0

**Rule:** [LawOfDemeter](https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter)

**Description:**

PMD report the method chain call (rule **LawOfDemeter**) with `softly.assertThat("doesn't matter").isEqualTo("doesn't matter")`, but when using `this.softly.assertThat("doesn't matter").isEqualTo("doesn't matter")`, it does not report.

**Code Sample demonstrating the issue:**
Maybe a false negative:
```java
import org.assertj.core.api.JUnitSoftAssertions;
import org.junit.Test;

public class FooTest {

    @Rule
    public final JUnitSoftAssertions softly = new JUnitSoftAssertions();

    @Test
    void testFoo() {
        this.softly.assertThat("doesn't matter").isEqualTo("doesn't matter");  // NOT report here
    }
}
```

Correctly reported code:
```java
import org.assertj.core.api.JUnitSoftAssertions;
import org.junit.Test;

public class FooTest {

    @Rule
    public final JUnitSoftAssertions softly = new JUnitSoftAssertions();

    @Test
    void testFoo() {
        softly.assertThat("doesn't matter").isEqualTo("doesn't matter");  // report here
    }
}

```

**Expected outcome:**
``` 
 LawOfDemeter:        Potential violation of Law of Demeter (method chain calls)
```

PMD should report a violation at line 11, but doesn't. This is a false negative.

**Running PMD through:** *[CLI ]*

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
